src/main/java/org/commcare/cases/entity/AsyncEntity.java

```diff
@@ -129,56 +131,50 @@ public String getNormalizedField(int i) {
 
     @Override
     public String getSortField(int i) {
-        try {
-            if (mEntityStorageCache == null || mEntityStorageCache.lockCache()) {
-                //get our second lock.
-                synchronized (mAsyncLock) {
-                    if (sortData[i] == null) {
-                        // sort data not in search field cache; load and store it
-                        Text sortText = fields[i].getSort();
-                        if (sortText == null) {
-                            return null;
-                        }
-
-                        String cacheKey = null;
-                        if (mEntityStorageCache != null) {
-                            cacheKey = mEntityStorageCache.getCacheKey(mDetailId, String.valueOf(i));
-
-                            if (mCacheIndex != null) {
-                                //Check the cache!
-                                String value = mEntityStorageCache.retrieveCacheValue(mCacheIndex, cacheKey);
-                                if (value != null) {
-                                    this.setSortData(i, value);
-                                    return sortData[i];
-                                }
+        try (Closeable ignored = mEntityStorageCache != null ? mEntityStorageCache.lockCache() : null) {

```

Comment:
unfortunately diff is super bad here because of indentation changes caused by `if` removal but the only effective change here is the replacement of -
````
try {
  if(mEntityStorageCache == null || mEntityStorageCache.lockCache())
````

by 

````
try (Closeable ignored = mEntityStorageCache != null ? mEntityStorageCache.lockCache() : null) {

````

in accordance with the [commit here](https://github.com/dimagi/commcare-core/pull/1387/commits/ca1b6f49bb5171b5cd12ab8b1ef56fe1c2587a32) 
````